service/authorization/authorization.go

```diff
@@ -399,61 +401,73 @@ func (as *AuthorizationService) GetEntitlements(ctx context.Context, req *author
-		if err != nil {
-			as.logger.ErrorContext(ctx, "failed to build rego input", slog.Any("entity", entity), slog.String("error", err.Error()))
-			return nil, status.Error(codes.Internal, "failed to build rego input")
-		}
-		as.logger.DebugContext(ctx, "entitlements", "entity_id", entity.GetId(), "input", fmt.Sprintf("%+v", in))
+	// call ERS on all entities
+	ersResp, err := as.sdk.EntityResoution.ResolveEntities(ctx, &entityresolution.ResolveEntitiesRequest{Entities: req.GetEntities()})
+	if err != nil {
+		as.logger.ErrorContext(ctx, "error calling ERS to resolve entities", "entities", req.GetEntities())
+		return nil, err
+	}
 
-		results, err := as.eval.Eval(ctx,
-			rego.EvalInput(in),
-		)
-		if err != nil {
-			return nil, status.Error(codes.Internal, "failed to evaluate entitlements policy")
-		}
+	// call rego on all entities
+	in, err := entitlements.OpaInput(subjectMappings, ersResp)
+	if err != nil {
+		as.logger.ErrorContext(ctx, "failed to build rego input", slog.String("error", err.Error()))
+		return nil, status.Error(codes.Internal, "failed to build rego input")
+	}
+	as.logger.DebugContext(ctx, "entitlements", "input", fmt.Sprintf("%+v", in))
 
-		// If we get no results and no error then we assume that the entity is not entitled to anything
-		if len(results) == 0 {
-			as.logger.DebugContext(ctx, "no entitlement results", slog.String("entity_id", entity.GetId()))
-			return rsp, nil
-		}
+	results, err := as.eval.Eval(ctx,
+		rego.EvalInput(in),
+	)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to evaluate entitlements policy")
+	}
+	// If we get no results and no error then we assume that the entity is not entitled to anything
+	if len(results) == 0 {
+		as.logger.DebugContext(ctx, "no entitlement results")
+		return rsp, nil
+	}
 
-		// I am not sure how we would end up with multiple results but lets return an empty entitlement set for now
-		if len(results) > 1 {
-			as.logger.WarnContext(ctx, "multiple entitlement results", slog.String("entity_id", entity.GetId()), slog.String("results", fmt.Sprintf("%+v", results)))
-			return rsp, nil
-		}
+	// I am not sure how we would end up with multiple results but lets return an empty entitlement set for now
+	if len(results) > 1 {
+		as.logger.WarnContext(ctx, "multiple entitlement results", slog.String("results", fmt.Sprintf("%+v", results)))
+		return rsp, nil
+	}
 
-		// If we get no expressions then we assume that the entity is not entitled to anything
-		if len(results[0].Expressions) == 0 {
-			as.logger.WarnContext(ctx, "no entitlement expressions", slog.String("entity_id", entity.GetId()), slog.String("results", fmt.Sprintf("%+v", results)))
-			return rsp, nil
-		}
+	// If we get no expressions then we assume that the entity is not entitled to anything
+	if len(results[0].Expressions) == 0 {
+		as.logger.WarnContext(ctx, "no entitlement expressions", slog.String("results", fmt.Sprintf("%+v", results)))
```

Comment:
This is less relevant for warnings (usually printed) than Debugs (usually not), but generally you want to defer serialization for values in case they are not ever done

```suggestion
		as.logger.WarnContext(ctx, "no entitlement expressions", slog.Any("results", results))
```

Reply:
we def need logging improvements across the service, theres issue for it, hoping to tackle this on engineering day https://github.com/opentdf/platform/issues/1235